src/basic/path-util.c

```diff
@@ -727,10 +727,7 @@ bool path_is_safe(const char *p) {
                 return false;
 
         /* The following two checks are not really dangerous, but hey, they still are confusing */
-        if (startswith(p, "./") || endswith(p, "/.") || strstr(p, "/./"))
-                return false;
-
-        if (strstr(p, "//"))
+        if (startswith(p, "./") || endswith(p, "/."))
```

Comment:
uh? why this change? as the comment says, we check here not only for things that are actively dangerous, but also confusing. I am not seeing why "/./" would ever be a good thing to permit here...